sdk/formrecognizer/Azure.AI.FormRecognizer/src/RecognizeContentOperation.cs

```diff
@@ -12,22 +12,62 @@
 
 namespace Azure.AI.FormRecognizer.Models
 {
-    internal class ExtractLayoutOperation : Operation<IReadOnlyList<ExtractedLayoutPage>>
+    /// <summary>
+    /// Tracks the status of a long-running operation for recognizing layout elements from forms.
+    /// </summary>
+    public class RecognizeContentOperation : Operation<IReadOnlyList<ExtractedLayoutPage>>
     {
+        /// <summary>Provides communication with the Form Recognizer Azure Cognitive Service through its REST API.</summary>
+        private readonly ServiceClient _serviceClient;
+
+        /// <summary>The last HTTP response received from the server. <c>null</c> until the first response is received.</summary>
         private Response _response;
+
+        /// <summary>The result of the long-running operation. <c>null</c> until result is received on status update.</summary>
         private IReadOnlyList<ExtractedLayoutPage> _value;
-        private bool _hasCompleted;
 
-        private readonly ServiceClient _operations;
+        /// <summary><c>true</c> if the long-running operation has completed. Otherwise, <c>false</c>.</summary>
+        private bool _hasCompleted;
 
+        /// <inheritdoc/>
         public override string Id { get; }
 
+        /// <inheritdoc/>
         public override IReadOnlyList<ExtractedLayoutPage> Value => OperationHelpers.GetValue(ref _value);
 
+        /// <inheritdoc/>
         public override bool HasCompleted => _hasCompleted;
 
+        /// <inheritdoc/>
         public override bool HasValue => _value != null;
 
+        /// <summary>
+        /// Initializes a new instance of the <see cref="RecognizeContentOperation"/> class.
+        /// </summary>
+        /// <param name="operationId">The ID of this operation.</param>
+        /// <param name="client">The client used to check for completion.</param>
+        public RecognizeContentOperation(string operationId, FormRecognizerClient client)
```

Comment:
where are we using this? I might be missing it somewhere but can't find it.
If we are, I am wondering if we can make a common internal constructor and have both of this call that one.
That common one can handle the argument validation and whatever else is needed

Reply:
This constructor is intended to be used by developers (we don't need to call it in the client). The flow would be something like this:
- The user creates a `FormRecognizerClient` and calls `StartRecognizeContent`. The method will create a `RecognizeContentOperation` with its `internal` constructor and return it to the user.
- The user can grab the id by calling `operation.Id`, store it and, some time later, they can create a new `RecognizeContentOperation` instance with the stored id (using the `public` constructor). They should be able to retrieve information about the same long-running operation from the new instance.

This is a pattern described in our guidelines. I'm not sure if there were long-running operations in the Text Analytics client, so you may already know about it, but I think this information could be useful to whoever reads your question.

Reply:
Regarding the common constructor for argument validation, I'm not sure if that's feasible since they have different parameters. Dealing with the client would be easy, but one of them accepts an `operationId` (I believe it's a GUID), and the other takes an `operationLocation`, which is an address to send requests to the service (some.thing/here/id). Sadly we handle them differently, but let me know if you have any ideas!

Reply:
ohh cool. Thanks for the explanation!
TA didn't have LRO so all of this is new to me :)

Reply:
BTW this is not critical as we can always change it, but I was thinking: we could at least change the order of the parameters so they are consistent. First client, then string, or other way around.
If more common things are needed, then we can add the common ctr